.github/workflows/ci.yml

```diff
@@ -79,3 +78,19 @@ jobs:
         shell: bash
         working-directory: PyOP2
         run: pytest test -v --tb=native
+
+      - name: Build documentation
+        if: ${{ github.event_name == 'push' }}
+        shell: bash
+        working-directory: PyOP2
+        run: make -C doc/sphinx html
+
+      - name: Upload to GitHub Pages
+        if: ${{ github.event_name == 'push' }}
```

Comment:
Can we guard this on pushes to master only as well?

Reply:
This workflow only gets triggered on either pull requests or pushes to master.